### PR TITLE
Update JAX to use XLA hyperbolic functions.

### PR DESCRIPTION
**Pull request by srvasude**
Use exposed hyperbolic functions from XLA in JAX.

This has the benefit of improved numerics for Sinh, Asinh and Atanh when x is close to zero. I've decided to do this vs. replicating the trick so that JAX and other XLA clients can benefit.

### CODE REVIEW

**Comment by srvasude** [comment]
Yikes I realize this is 27 commits. My github/git knowledge is a little lacking. Anyway I can condense this / remove most of these?

**Comment by hawkinsp** [comment]
You can either use `git rebase`  to squash those commits, or you can ask the person merging to use Github's "squash and merge" option.

It looks like there is a real test failure in -x64 mode?

**Comment by srvasude** [comment]
Tests should be fixed. I changed uses of 1. to _one(x) which fixed the issue.

File: tests/lax_test.py [hunk 1]
```diff
@@ -112,6 +112,8 @@ def op_record(op, nargs, dtypes, rng_factory, tol=None):
     op_record("asin", 1, float_dtypes, jtu.rand_small),
     op_record("acos", 1, float_dtypes, jtu.rand_small),
     op_record("atan", 1, float_dtypes, jtu.rand_small),
+    op_record("asinh", 1, float_dtypes, jtu.rand_default),
+    op_record("acosh", 1, float_dtypes, jtu.rand_positive),
```
Comment (hawkinsp) [inline tests/lax_test.py]: Can you add a test for `atanh` here too?

Otherwise LGTM.

**Comment by srvasude** [comment]
FWIW, the test failure seems unrelated.

**Comment by mattjj** [comment]
I agree it seems unrelated. Could you mark the test as skipped for now (just raise SkipTest, as other test cases do), and @froystig can you follow up on the error?

**Comment by srvasude** [comment]
Sounds good. Disabled the test, and added a TODO.

**Pull request merged**